stylelint-lsp: fix script launch issue on macOS

growler opened the pull request:
A leading space before the shebang doesn't affect script execution when run from the shell
or via exec on Linux. However, on macOS, it results in an ENOEXEC (File format error) when using exec.
This fix removes the space to ensure stylelint-lsp can be launched from nixvim on macOS without errors.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

torgeirsh committed:
rdkafka: Add curl dependency for OAuth support.

jackwilsdon committed:
plexamp: add ozone platform hint for wayland

chreekat committed:
nixos/github-runners: make enable functional

Fixes #305304

Aleksanaa committed:
gnome-graphs: 1.8.1 -> 1.8.2

aanderse committed:
nixos/userborn: fix username typo

errnoh committed:
coroot: init at 1.5.8

vbgl committed:
ocamlPackages.merlin: 5.1-502 → 5.2.1-502

lukts30 committed:
write_stylus: rename to styluslabs-write-bin

felixsinger committed:
nixos/tests/gerrit: Drop dead hook to LFS plugin

The URL scheme for downloading plugins has changed a long time ago and
the used URL is dead. Gerrit only throws an error since it can't load
the plugin but it continues to boot. However, instead of maintaining
URLs to 3rdparty plugins, which end up dead anyway, just drop it. The
test should cover Gerrit and not 3rd party plugins.

Also, while on it, drop the setting `plugins.allowRemoteAdmin = true`
since it's not needed.

Signed-off-by: Felix Singer <felixsinger@posteo.net>

FliegendeWurst committed:
ccid: 1.5.5 -> 1.6.1

markuskowa committed:
amd-libflame: 4.2 -> 5.0

markuskowa committed:
aocl-utils: 4.2 -> 5.0

markuskowa committed:
aocl-utils: apply nixfmt

markuskowa committed:
amd-libflame: apply nixfmt

dotlambda committed:
python312Packages.mat2: disable failing test

r-ryantm committed:
nb: 7.12.1 -> 7.14.4

Steinhagen committed:
lutris: fix libjansson and cursor issues

This fixes the following warning:
ntlm_auth: /home/reloner/.local/share/lutris/runtime/Ubuntu-18.04-x86_64/libjansson.so.4: no version information available (required by /nix/store/4gcggjvzrbywi92r66p6l15hckxng2q0-samba-4.20.1/lib/samba/libcommon-auth-private-samba.so)
The reason why this issue is happening is because the Lutris FHS that you are running doesn't install jansson as a dependency. By default the WoW entry tries to prioritize the System's libraries and if it doesn't find something, it will fallback to the Lutris runtime. Since the runtime version is really old, you'll see that error.

This commit also fixes cursor issues caused by xorg in WoW.

r-ryantm committed:
eask-cli: 0.10.0 -> 0.10.1

r-ryantm committed:
homebank: 5.8.3 -> 5.8.5

numinit committed:
mattermost: 9.5.10 -> 9.5.11

Fixes MMSA-2024-00370; see:
https://mattermost.com/security-updates/

numinit committed:
{mattermost,matterircd}: relocate to by-name

r-ryantm committed:
vpl-gpu-rt: 24.2.5 -> 24.3.3

u2d3rsc0re committed:
tlp: 1.6.1 -> 1.7.0

hellodword committed:
sing-box: 1.9.7 -> 1.10.0

r-ryantm committed:
marwaita-icons: 5.0 -> 5.1

NickCao committed:
stratisd: 3.7.1 -> 3.7.2

Diff: https://github.com/stratis-storage/stratisd/compare/refs/tags/stratisd-v3.7.1...stratisd-v3.7.2

aqrln committed:
prisma-engines: 5.18.0 -> 5.21.0

bjornfor committed:
maven: 3.9.8 -> 3.9.9

aqrln committed:
prisma: 5.18.0 -> 5.21.0

HeitorAugustoLN committed:
audacity: move to by-name

HeitorAugustoLN committed:
audacity: format with nixfmt-rfc-style

HeitorAugustoLN committed:
audacity: remove `with lib` from meta and `rec`

r-ryantm committed:
pinentry-rofi: 2.2.0 -> 3.0.0

lukts30 committed:
styluslabs-write: init at 2024-10-12

r-ryantm committed:
python312Packages.cvelib: 1.5.0 -> 1.6.0

khaneliman committed:
cliphist: 0.5.0 -> 0.6.0

khaneliman committed:
cliphist: move to by-name

khaneliman committed:
cliphist: add update script

skeuchel committed:
alsa-firmware: fix riscv64-linux build

dotlambda committed:
python312Packages.sense-energy: 0.12.4 -> 0.13.2

Diff: https://github.com/scottbonline/sense/compare/refs/tags/0.12.4...0.13.2

Changelog: https://github.com/scottbonline/sense/releases/tag/0.13.0
           https://github.com/scottbonline/sense/releases/tag/0.13.1
           https://github.com/scottbonline/sense/releases/tag/0.13.2

dotlambda committed:
python312Packages.onnx: clean up dependencies

dotlambda committed:
python312Packages.onnx: 1.16.2 -> 1.17.0

Diff: https://github.com/onnx/onnx/compare/refs/tags/v1.16.2...v1.17.0

Changelog: https://github.com/onnx/onnx/releases/tag/v1.17.0

NickCao committed:
telegram-desktop: 5.6.2 -> 5.6.3

Diff: https://github.com/telegramdesktop/tdesktop/compare/f2e0e481de5d379b9542f6ff9021fab5d409c1d4...v5.6.3

Changelog: https://github.com/telegramdesktop/tdesktop/releases/tag/v5.6.3

fabaff committed:
python312Packages.tencentcloud-sdk-python: 3.0.1249 -> 3.0.1250

Diff: https://github.com/TencentCloud/tencentcloud-sdk-python/compare/refs/tags/3.0.1249...3.0.1250

Changelog: https://github.com/TencentCloud/tencentcloud-sdk-python/blob/3.0.1250/CHANGELOG.md

NickCao committed:
v2ray-domain-list-community: 20240920063125 -> 20241013063848

Diff: https://github.com/v2fly/domain-list-community/compare/20240920063125...20241013063848

fabaff committed:
python312Packages.opower: 0.8.3 -> 0.8.4

Diff: https://github.com/tronikos/opower/compare/refs/tags/v0.8.3...v0.8.4

Changelog: https://github.com/tronikos/opower/releases/tag/v0.8.4

fabaff committed:
python312Packages.pydeconz: 117 -> 118

Diff: https://github.com/Kane610/deconz/compare/refs/tags/v117...v118

Changelog: https://github.com/Kane610/deconz/releases/tag/v118

fabaff committed:
python312Packages.aioftp: 0.22.3 -> 0.23.1

r-ryantm committed:
kustomize: 5.4.3 -> 5.5.0

pyrox0 committed:
python312Packages.androguard: Remove unused dependencies

pyrox0 committed:
python312Packages.androguard: Remove mock dependency

pyrox0 committed:
python312Packages.androguard: Remove `with lib;`

khaneliman committed:
cliphist: 0.6.0 -> 0.6.1

r-ryantm committed:
python312Packages.gcovr: 7.2 -> 8.2

jchv committed:
linuxPackages.lttng-modules: 2.13.13 -> 2.13.15

Fixes the build for linuxPackages_latest.lttng-modules.

jchv committed:
linuxPackages.lttng-modules: reformat with RFC style

r-ryantm committed:
python312Packages.stdlib-list: 0.10.0 -> 0.11.0

github-actions[bot] committed:
terraform-providers.aiven: 4.24.0 -> 4.27.0

github-actions[bot] committed:
terraform-providers.akamai: 6.4.0 -> 6.5.0

github-actions[bot] committed:
terraform-providers.artifactory: 11.9.1 -> 12.2.0

github-actions[bot] committed:
terraform-providers.alicloud: 1.230.0 -> 1.231.0

github-actions[bot] committed:
terraform-providers.auth0: 1.6.0 -> 1.7.1

github-actions[bot] committed:
terraform-providers.argocd: 6.1.1 -> 6.2.0

github-actions[bot] committed:
terraform-providers.aviatrix: 3.1.5 -> 3.2.0

github-actions[bot] committed:
terraform-providers.azuread: 2.53.1 -> 3.0.2

github-actions[bot] committed:
terraform-providers.bigip: 1.22.3 -> 1.22.4

github-actions[bot] committed:
terraform-providers.bitwarden: 0.8.1 -> 0.10.0

github-actions[bot] committed:
terraform-providers.azurerm: 4.1.0 -> 4.5.0

github-actions[bot] committed:
terraform-providers.buildkite: 1.10.2 -> 1.12.0

github-actions[bot] committed:
terraform-providers.aws: 5.66.0 -> 5.72.0

github-actions[bot] committed:
terraform-providers.cloudinit: 2.3.4 -> 2.3.5

github-actions[bot] committed:
terraform-providers.digitalocean: 2.40.0 -> 2.42.0

github-actions[bot] committed:
terraform-providers.dns: 3.4.1 -> 3.4.2

github-actions[bot] committed:
terraform-providers.datadog: 3.44.0 -> 3.46.0

github-actions[bot] committed:
terraform-providers.doppler: 1.10.0 -> 1.11.0

github-actions[bot] committed:
terraform-providers.exoscale: 0.59.2 -> 0.61.0

github-actions[bot] committed:
terraform-providers.equinix: 2.4.1 -> 2.8.0

github-actions[bot] committed:
terraform-providers.external: 2.3.3 -> 2.3.4

github-actions[bot] committed:
terraform-providers.fastly: 5.13.0 -> 5.14.0

github-actions[bot] committed:
terraform-providers.github: 6.2.3 -> 6.3.1

github-actions[bot] committed:
terraform-providers.gitlab: 17.3.1 -> 17.4.0

github-actions[bot] committed:
terraform-providers.fortios: 1.20.0 -> 1.21.0

github-actions[bot] committed:
terraform-providers.google: 6.2.0 -> 6.7.0

github-actions[bot] committed:
terraform-providers.grafana: 3.7.0 -> 3.10.0

github-actions[bot] committed:
terraform-providers.google-beta: 6.2.0 -> 6.7.0

github-actions[bot] committed:
terraform-providers.http: 3.4.4 -> 3.4.5

github-actions[bot] committed:
terraform-providers.helm: 2.15.0 -> 2.16.1

github-actions[bot] committed:
terraform-providers.huaweicloud: 1.68.1 -> 1.69.0

github-actions[bot] committed:
terraform-providers.libvirt: 0.7.6 -> 0.8.0

github-actions[bot] committed:
terraform-providers.kubernetes: 2.32.0 -> 2.33.0

github-actions[bot] committed:
terraform-providers.ibm: 1.69.0 -> 1.70.0

github-actions[bot] committed:
terraform-providers.linode: 2.27.0 -> 2.29.1

github-actions[bot] committed:
terraform-providers.local: 2.5.1 -> 2.5.2

github-actions[bot] committed:
terraform-providers.minio: 2.5.0 -> 2.5.1

github-actions[bot] committed:
terraform-providers.lxd: 2.3.0 -> 2.4.0

github-actions[bot] committed:
terraform-providers.migadu: 2024.9.5 -> 2024.10.10

github-actions[bot] committed:
terraform-providers.mongodbatlas: 1.18.1 -> 1.21.1

github-actions[bot] committed:
terraform-providers.newrelic: 3.45.0 -> 3.50.0

github-actions[bot] committed:
terraform-providers.nomad: 2.3.1 -> 2.4.0

github-actions[bot] committed:
terraform-providers.ns1: 2.4.1 -> 2.4.4

github-actions[bot] committed:
terraform-providers.null: 3.2.2 -> 3.2.3

github-actions[bot] committed:
terraform-providers.openstack: 2.1.0 -> 3.0.0

github-actions[bot] committed:
terraform-providers.okta: 4.10.0 -> 4.11.0

github-actions[bot] committed:
terraform-providers.opentelekomcloud: 1.36.18 -> 1.36.20

github-actions[bot] committed:
terraform-providers.oci: 6.9.0 -> 6.13.0

github-actions[bot] committed:
terraform-providers.ovh: 0.48.0 -> 0.51.0

github-actions[bot] committed:
terraform-providers.pagerduty: 3.15.6 -> 3.16.0

r-ryantm committed:
snips-sh: 0.4.0 -> 0.4.1

github-actions[bot] committed:
terraform-providers.project: 1.7.2 -> 1.8.0

github-actions[bot] committed:
terraform-providers.random: 3.6.2 -> 3.6.3

github-actions[bot] committed:
terraform-providers.rundeck: 0.4.7 -> 0.4.9

github-actions[bot] committed:
terraform-providers.rancher2: 5.0.0 -> 5.1.0

github-actions[bot] committed:
terraform-providers.scaleway: 2.44.0 -> 2.46.0

github-actions[bot] committed:
terraform-providers.selectel: 5.3.0 -> 5.4.0

github-actions[bot] committed:
terraform-providers.spacelift: 1.15.0 -> 1.16.1

github-actions[bot] committed:
terraform-providers.spotinst: 1.190.0 -> 1.195.0

github-actions[bot] committed:
terraform-providers.snowflake: 0.95.0 -> 0.97.0

github-actions[bot] committed:
terraform-providers.tailscale: 0.16.2 -> 0.17.2

github-actions[bot] committed:
terraform-providers.sumologic: 2.31.3 -> 2.31.5

github-actions[bot] committed:
terraform-providers.temporalcloud: 0.0.11 -> 0.0.13

github-actions[bot] committed:
terraform-providers.talos: 0.5.0 -> 0.6.0

github-actions[bot] committed:
terraform-providers.tfe: 0.58.1 -> 0.59.0

github-actions[bot] committed:
terraform-providers.time: 0.12.0 -> 0.12.1

github-actions[bot] committed:
terraform-providers.tencentcloud: 1.81.120 -> 1.81.133

github-actions[bot] committed:
terraform-providers.tls: 4.0.5 -> 4.0.6

github-actions[bot] committed:
terraform-providers.turbot: 1.11.1 -> 1.11.2

github-actions[bot] committed:
terraform-providers.vcd: 3.13.0 -> 3.14.0

github-actions[bot] committed:
terraform-providers.venafi: 0.21.0 -> 0.21.1

github-actions[bot] committed:
terraform-providers.vsphere: 2.9.1 -> 2.9.3

github-actions[bot] committed:
terraform-providers.yandex: 0.128.0 -> 0.130.0

r-ryantm committed:
vrc-get: 1.8.1 -> 1.8.2

zowoq committed:
Revert "terraform-providers: bump go to 1.23"

1.23 is now the default

This reverts commit d67593d858e1f759246089cde74592773974578e.

zowoq committed:
.github/workflows: remove update-terraform-providers

semi-broken, will try using r-ryantm bot for updates instead

zowoq committed:
terraform-providers: enable update bot

r-ryantm committed:
tdnf: 3.5.8 -> 3.5.9

tomodachi94 committed:
{nodePackages,vimPlugins}.coc-python: drop

See notice in the README:
https://github.com/neoclide/coc-python

> WARNING: it's recommended to use coc-pyright if
> you're using python3 or use coc-jedi if you're using jedi,
> the code of coc-python is too hard to maintain!

If that isn't convincing, the repo was archived on 2020-12-24.

Part of #229475

iynaix committed:
wallust: move to by-name

iynaix committed:
wallust: 3.0.0 -> 3.1.0

Changelog: https://codeberg.org/explosion-mental/wallust/releases/tag/3.1.0
Diff: https://codeberg.org/explosion-mental/wallust/compare/3.0.0...3.1.0

growler committed:
stylelint-lsp: fix script launch issue on macOS

A leading space before the shebang doesn't affect script execution
when run from the shell or via exec on Linux. However, on macOS,
it results in an ENOEXEC (File format error) when using exec.
This fix removes the space to ensure stylelint-lsp can be launched
from nixvim on macOS without errors.

K900 committed:
buildFHSEnv: don't include /var

This mostly prevents random junk from leaking into fhsenvs,
but also fixes Steam not having a /var anymore.

fabaff committed:
python312Packages.gcovr: refactor

GaetanLepage committed:
python312Packages.onnx: 1.16.2 -> 1.17.0

Diff: https://github.com/onnx/onnx/compare/refs/tags/v1.16.2...v1.17.0

Changelog: https://github.com/onnx/onnx/releases/tag/v1.17.0

msanft committed:
erofs-utils: 1.8.1 -> 1.8.2

r-ryantm committed:
tiny-dfr: 0.3.1 -> 0.3.2

r-ryantm committed:
soju: 0.8.1 -> 0.8.2

VergeDX committed:
qv2ray: fix build

GaetanLepage committed:
python312Packages.nvidia-ml-py: fix and improve test

Aleksanaa committed:
gitmux: fix build

kachick committed:
sequoia-chameleon-gnupg: unstable-2023-11-22 -> 0.11.2

r-ryantm committed:
python312Packages.pyoverkiz: 1.14.1 -> 1.14.2

r-ryantm committed:
python312Packages.libknot: 3.4.0 -> 3.4.1

r-ryantm committed:
seafile-shared: 9.0.7 -> 9.0.8

fabaff committed:
python312Packages.tencentcloud-sdk-python: 3.0.1250 -> 3.0.1251

Diff: https://github.com/TencentCloud/tencentcloud-sdk-python/compare/refs/tags/3.0.1250...3.0.1251

Changelog: https://github.com/TencentCloud/tencentcloud-sdk-python/blob/3.0.1251/CHANGELOG.md

r-ryantm committed:
tbls: 1.77.0 -> 1.78.0

r-ryantm committed:
cargo-deb: 2.5.1 -> 2.7.0

r-ryantm committed:
libcdada: 0.6.0 -> 0.6.1

fabaff committed:
cnspec: 11.25.0 -> 11.26.0

Diff: https://github.com/mondoohq/cnspec/compare/refs/tags/v11.25.0...v11.26.0

Changelog: https://github.com/mondoohq/cnspec/releases/tag/v11.26.0

gepbird commented on pkgs/by-name/st/stylelint-lsp/package.nix:
```diff
@@ -41,9 +41,9 @@ stdenvNoCC.mkDerivation (finalAttrs: {
-    echo "
-      #!${lib.getExe bash}
-      ${lib.getExe nodejs} $out/lib/${finalAttrs.pname}/dist/index.js \$@
+    echo "\
+    #!${lib.getExe bash}
+    ${lib.getExe nodejs} $out/lib/${finalAttrs.pname}/dist/index.js \$@
```
A bit nicer way to do it would be using `wrapProgram`.
For that to work, we also need to mark the source js file as executable.
```suggestion
    chmod a+x $out/lib/${finalAttrs.pname}/dist/index.js
    makeWrapper $out/lib/${finalAttrs.pname}/dist/index.js $out/bin/stylelint-lsp  \
      --prefix "${lib.getExe nodejs}"
```

growler replied:
Sorry I screwed up the PR, will close it and fill a new one with a suggested fix.

gepbird reviewed (changes requested):
Thanks for your first contribution!

You're right, there are unnecessary spaces and newlines in the bootstrap script.

mweinelt committed:
prometheus-knot-exporter: 3.4.0 -> 3.4.1

hellodword committed:
nixos/sing-box: migration of deprecated fields

4evy committed:
google-chrome: fix URL encoding for version filter

4evy committed:
google-chrome: add `shellcheck`

4evy committed:
google-chrome: 129.0.6668.100 -> 130.0.6723.58

Changelog: https://chromereleases.googleblog.com/2024/10/stable-channel-update-for-desktop_15.html

4evy committed:
google-chrome: 129.0.6668.101 -> 130.0.6723.59

Changelog: https://chromereleases.googleblog.com/2024/10/stable-channel-update-for-desktop_15.html

mweinelt committed:
epshome: 2024.9.2 -> 2024.10.0

https://github.com/esphome/esphome/releases/tag/2024.10.0

r-ryantm committed:
vscode-extensions.pylyzer.pylyzer: 0.1.8 -> 0.1.10

fabaff committed:
python312Packages.pyoverkiz: update disabled

corngood committed:
jam: fix OS determination on linux

corngood committed:
jam: add test for OS=UNKNOWN

hellodword committed:
nixos/sing-box: test iproute2_table_index and iproute2_rule_index

r-ryantm committed:
mlx42: 2.3.4 -> 2.4.0

K900 committed:
obs-vkcapture: don't depend on ECM

Fixes #349078.

adamcstephens committed:
nixos/atticd: wants network-online.target

fixes:

trace: evaluation warning: atticd.service is ordered after 'network-online.target' but doesn't depend on it

r-ryantm committed:
sqlitebrowser: 3.13.0 -> 3.13.1

growler committed:
stylelint-lsp: fix script launch issue on macOS

A proper fix as suggested by @gepbird
